modules/geochemistry/src/utils/GeochemistryActivityCoefficients.C

```diff
@@ -0,0 +1,180 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "GeochemistryActivityCoefficients.h"
+#include "GeochemistryActivityCalculators.h"
+
+GeochemistryActivityCoefficients::GeochemistryActivityCoefficients(
+    ActivityCoefficientMethodEnum method, const GeochemistryIonicStrength & is_calculator)
+  : _method(method),
+    _is_calculator(is_calculator),
+    _ionic_strength(1.0),
+    _sqrt_ionic_strength(1.0),
+    _stoichiometric_ionic_strength(1.0),
+    _num_basis(0),
+    _num_eqm(0),
+    _dhA(0.0),
+    _dhB(0.0),
+    _dhBdot(0.0),
+    _dha(0.0),
+    _dhb(0.0),
+    _dhc(0.0),
+    _dhd(0.0),
+    _dhatilde(0.0),
+    _dhbtilde(0.0),
+    _dhctilde(0.0),
+    _dhdtilde(0.0)
+{
+}
+
+void
+GeochemistryActivityCoefficients::setInternalParameters(
+    Real /*temperature*/,
+    const ModelGeochemicalDatabase & mgd,
+    const std::vector<Real> & basis_species_molality,
+    const std::vector<Real> & eqm_species_molality,
+    const std::vector<Real> & kin_species_molality)
+{
+  _ionic_strength = _is_calculator.ionicStrength(
+      mgd, basis_species_molality, eqm_species_molality, kin_species_molality);
+  _sqrt_ionic_strength = std::sqrt(_ionic_strength);
+  _stoichiometric_ionic_strength = _is_calculator.stoichiometricIonicStrength(
+      mgd, basis_species_molality, eqm_species_molality, kin_species_molality);
+  _num_basis = mgd.basis_species_index.size();
+  _num_eqm = mgd.eqm_species_index.size();
+  // when temperature least-squares is completed the following lines will be changed
+  // charged species Debye-Huckel at 25degC
+  _dhA = 0.5092;
+  _dhB = 0.3283;
+  _dhBdot = 0.035;
+  // water Debye-Huckel at 25degC
+  _dhatilde = 1.45397;
+  _dhbtilde = 0.022357;
+  _dhctilde = 0.0093804;
+  _dhdtilde = -0.0005262;
+  // neutral species Debye-Huckel at 25degC
+  _dha = 0.1127;
+  _dhb = -0.01049;
+  _dhc = 0.001545;
+  _dhd = 0.0;
+}
```

Comment:
Oh, I see. Waiting until you can interpolate them.

Reply:
yep :-)